cairo/ethereum/cancun/trie.cairo

```diff
@@ -976,28 +962,22 @@ func _get_branches{poseidon_ptr: PoseidonBuiltin*}(obj: MappingBytesBytes, level
 
 // @notice Given a key (inside `dict_ptr`), returns the preimage of the key registered in the tracker.
 // The preimage is validated to be correctly provided by the prover by hashing it and comparing it to the key.
+// @param key - The key to get the preimage for. Either a hashed or non-hashed key - but it must be a felt.
+// @param dict_ptr_stop - The pointer to the end of the dict segment, the one registered in the tracker.
 func _get_preimage_for_key{poseidon_ptr: PoseidonBuiltin*}(
-    dict_ptr: BytesBytesDictAccess*, dict_ptr_stop: BytesBytesDictAccess*
+    key: felt, dict_ptr_stop: BytesBytesDictAccess*
 ) -> Bytes {
     alloc_locals;
 
     // Get preimage data
     let (local preimage_data: felt*) = alloc();
     local preimage_len;
-    %{
-        from starkware.cairo.lang.vm.crypto import poseidon_hash_many
-        hashed_value = ids.dict_ptr.key.value
-        dict_tracker = __dict_manager.get_tracker(ids.dict_ptr_stop)
-        # Get the key in the dict that matches the hashed value
-        preimage = bytes(next(key for key in dict_tracker.data.keys() if poseidon_hash_many(key) == hashed_value))
-        segments.write_arg(ids.preimage_data, preimage)
-        ids.preimage_len = len(preimage)
-    %}
+    %{ get_preimage_for_key %}
```

Comment:
I really think that not only this refacto better for the devx, but it's also much cleaner when the hint content is not visible so that one directly catch what's in the program and what's not

<img width="503" alt="image" src="https://github.com/user-attachments/assets/bdbab900-a85c-4237-8be0-305dbb1347d9" />

Reply:
totally agree with that :)